sia/src/types/currency.rs

```diff
@@ -26,8 +23,54 @@ impl Serialize for Currency {
 
 impl<'de> Deserialize<'de> for Currency {
     fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
-        let s = String::deserialize(deserializer)?;
-        Currency::parse_string(&s).map_err(|e| serde::de::Error::custom(format!("{:?}", e)))
+        struct CurrencyVisitor;
+
+        impl serde::de::Visitor<'_> for CurrencyVisitor {
```

Comment:
If you pass `'de` as lifetime value here, wouldn't that work for both static and dynamically scoped input? 

```rust
struct CurrencyVisitor<'de>;

impl<'de> serde::de::Visitor<'de> for CurrencyVisitor<'de> {
    type Value = Currency;

    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
        formatter.write_str("a string or numeric representing a currency value")
    }

    fn visit_str<E: serde::de::Error>(self, s: &'de str) -> Result<Self::Value, E> {
        Currency::parse_string(s).map_err(|e| serde::de::Error::custom(format!("{:?}", e)))
    }

    // other visit methods
}

deserializer.deserialize_any(CurrencyVisitor)

```

Reply:
You can't use the lifetime of the outer `impl` in the inner one.

Reply:
:homer-bush-gif: